### PR TITLE
clever-tools 0.9.3

### PR DESCRIPTION
**Pull request by fmenou**
## What's in it

2 derivations to install
- nodegit.node 0.20.3 for node v51, from prebuilt archive
- clever-tools 0.9.3, from prebuilt archive

Main job of those derivations is to use patchelf to make it nix compliant.

## How to use it

```nix
{ pkgs ? import <nixpkgs> {} }:

let
  clever-tools = pkgs.fetchFromGitHub
                   { owner  = "fmenou";
                     repo   = "clever-tools-nix";
                     rev    = "7c2a7868b387ffd064dd14c51f7b18224cdc5c88";
                     sha256 = "0r4bc4k90xpq325djvsyix9a6hiabmd4nba14j0plgdrwgzb36rh";
                   };

in import clever-tools { inherit pkgs; }
```

### CODE REVIEW

**Comment by hussein-aitlahcen** [comment]
LGTM, the only improvement I would do is changing the directory architecture to `nodegit/default.nix` and `clever-tools/default.nix` so that we could use something more convenient like (as suggested [here](https://nixos.org/nixos/nix-pills/callpackage-design-pattern.html)):
```nix
let clever-tools = callPackage ./clever-tools {};
    nodegit      = callPackage ./nodegit {};
in {}
```
This is especially useful when you want to add a custom package to nixos as well as implicitly passing the arguments to the derivation.

**Review by hussein-aitlahcen — APPROVED** [review]

File: default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  nodegit = import ./nodegit.nix { inherit pkgs; };
+
+  libs = [ stdenv.cc.cc glibc ];
+
+in
+
+  stdenv.mkDerivation {
+    name = "clever-cli-${version}";
+
+    buildInputs = [ nodegit nodejs-6_x ];
```
Comment (fmenou) [inline default.nix]: I'm not sure about that nodejs dep.

Comment (hussein-aitlahcen) [reply default.nix]: *He surely made it on purpose*

yo

File: nodegit.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  stdenv.mkDerivation {
+    name = "nodegit-${version}";
+
+    src = fetchurl {
+      url = "https://nodegit.s3.amazonaws.com/nodegit/nodegit/nodegit-v${version}-node-v51-linux-x64.tar.gz";
+      sha256 = "d022a88e58fa70f78b59b475fa6ecc6724d372a779b132264cf571f71bc50020";
+    };
+
+    buildPhase = ":";
```
Comment (hussein-aitlahcen) [inline nodegit.nix]: What about that :) ?

File: default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  version = "0.9.3";
+
+  nodegit = import ./nodegit.nix { inherit pkgs; };
+
+  libs = [ stdenv.cc.cc glibc ];
+
+in
+
+  stdenv.mkDerivation {
+    name = "clever-cli-${version}";
```
Comment (hussein-aitlahcen) [inline default.nix]: I saw that you were importing the package as `clever-tools`.
Do we have a good reason to name it differently (`clever-cli`) here ?

Comment (fmenou) [reply default.nix]: No I don't. Remember I don't know what I'm doing here.

File: default.nix (hunk 1 above)
Comment (fmenou) [reply default.nix]: I have no clue. It made sense at the time, but I'm pretty sure we could remove it, and it should work with other node variant :thinking:

File: nodegit.nix (hunk 2 above)
Comment (fmenou) [reply nodegit.nix]: To be honest I've blindly followed https://nixos.wiki/wiki/Packaging/Binaries#Creating_the_Derivation_for_upstream_Packaging

File: default.nix [hunk 4]
```diff
@@ -12,7 +12,7 @@ let
 in
 
   stdenv.mkDerivation {
-    name = "clever-cli-${version}";
+    name = "clever-tools-${version}";
 
     buildInputs = [ nodegit nodejs-6_x ];
```
Comment (hussein-aitlahcen) [inline default.nix]: I just tried and everything run smoothly regardless of the nodejs-xx dependence. I guess we can remove it :P

**Review by hussein-aitlahcen — APPROVED** [review]

File: default.nix (hunk 4 above)
Comment (fmenou) [reply default.nix]: I'm testing on one of our project.

**Pull request merged**